app/src/main/java/com/m2049r/xmrwallet/data/BarcodeData.java

```diff
@@ -159,17 +159,22 @@ static public BarcodeData parseMoneroUri(String uri) {
         String noScheme = uri.substring(XMR_SCHEME.length());
         Uri monero = Uri.parse(noScheme);
         Map<String, String> parms = new HashMap<>();
-        String query = monero.getEncodedQuery();
-        if (query != null) {
-            String[] args = query.split("&");
-            for (String arg : args) {
-                String[] namevalue = arg.split("=");
-                if (namevalue.length == 0) {
-                    continue;
+        try {
+            String query = monero.getEncodedQuery();
+            if (query != null) {
+                String[] args = query.split("&");
+                for (String arg : args) {
+                    String[] namevalue = arg.split("=");
+                    if (namevalue.length == 0) {
+                        continue;
+                    }
+                    parms.put(Uri.decode(namevalue[0]).toLowerCase(),
+                            namevalue.length > 1 ? Uri.decode(namevalue[1]) : "");
                 }
-                parms.put(Uri.decode(namevalue[0]).toLowerCase(),
-                        namevalue.length > 1 ? Uri.decode(namevalue[1]) : "");
             }
+        } catch (Exception e) {
```

Comment:
What exception is it really? Can we be more specific here?

Reply:
I might find the time to find out, but I figured that preventing the crash was more important than adding details. My description in #685 explains more.

Reply:
yes, i read that - very detailed and interesting

Reply:
afaict ```getEncodedQuery``` isn't throwing anything - it crashes because ```address``` is ```null``` - #702 fixes that